internal/compiler/host.go

```diff
@@ -32,6 +33,10 @@ type compilerHost struct {
 	defaultLibraryPath string
 }
 
+func NewCachedFSCompilerHost(options *core.CompilerOptions, currentDirectory string, fs vfs.FS, defaultLibraryPath string) CompilerHost {
+	return NewCompilerHost(options, currentDirectory, cachedvfs.From(fs), defaultLibraryPath)
```

Comment:
When does this cached FS get cleared? I guess never for the main entrypoint is good, but I'll be curious how we handle this elsewhere. Maybe it'll just be "we throw away the host anyway" so it works out.

Probably we should update the `tsc` entrypoint too? That one's trickier because it's also writable...

Reply:
This host is used exclusively by tsgo/main, and I wasn't aware of `internal/execute`, so I'll take a look at that. I thought about making Program itself wrap its host's FS, but it would require a bit of refactoring to make Program the host for file parsing and module resolution.